velox/common/memory/MemoryPool.h

```diff
@@ -243,6 +255,52 @@ class MemoryPool : public std::enable_shared_from_this<MemoryPool> {
     VELOX_NYI("release() needs to be implemented in derived memory pool.");
   }
 
+  /// Memory arbitration related interfaces.
+
+  /// Returns the number of bytes that haven't been reserved for use, and can be
+  /// freed by shrinking this memory pool's limit.
+  virtual uint64_t freeBytes() const = 0;
+
+  /// Invoked to free up to the specified amount of memory by shrinking this
+  /// memory pool's limit. The function returns the actually freed memory bytes.
+  /// If 'targetBytes' is zero, the function free up all the unreserved memory
+  /// bytes.
+  virtual uint64_t shrink(uint64_t targetBytes = 0) = 0;
+
+  /// Invoked to bump up the memory pool's limit by 'bytes'. The function
```

Comment:
Does 'limit' here refer to Options.capacity? Should we rename Options.capacity to limit and update the comments? What does "memory capacity" currently means?

```
  struct Options {
    /// Specifies the memory allocation alignment through this memory pool.
    uint16_t alignment{MemoryAllocator::kMaxAlignment};
    /// Specifies the memory capacity of this memory pool.
    int64_t capacity{kMaxMemory};
  };
```

Reply:
Yes, let's unify the terms.